.github/workflows/preview-docs.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Script to manage documentation preview comments on pull requests.
+ * This replaces the Python script dev/preview_docs.py to eliminate the need for Python setup
+ * in the preview-docs workflow.
+ */
+
+const MARKER = "<!-- documentation preview -->";
+
+/**
+ * Create or update a PR comment with documentation preview information
+ * @param {object} github - GitHub API client
+ * @param {string} repo - Repository name in format "owner/repo"
+ * @param {string} pullNumber - Pull request number
+ * @param {string} commentBody - Comment body content
+ */
+async function upsertComment(github, repo, pullNumber, commentBody) {
+  const [owner, repoName] = repo.split("/");
+
+  // Get existing comments on the PR
+  const { data: comments } = await github.rest.issues.listComments({
+    owner,
+    repo: repoName,
+    issue_number: pullNumber,
+  });
+
+  // Find existing preview docs comment
+  const previewDocsComment = comments.find((comment) => comment.body.includes(MARKER));
+  const commentBodyWithMarker = `${MARKER}\n\n${commentBody}`;
+
+  if (!previewDocsComment) {
+    console.log("Creating comment");
+    await github.rest.issues.createComment({
+      owner,
+      repo: repoName,
+      issue_number: pullNumber,
+      body: commentBodyWithMarker,
+    });
+  } else {
+    console.log("Updating comment");
+    await github.rest.issues.updateComment({
+      owner,
+      repo: repoName,
+      comment_id: previewDocsComment.id,
+      body: commentBodyWithMarker,
+    });
+  }
+}
+
+/**
+ * Generate the comment template for documentation preview
+ * @param {string} commitSha - Git commit SHA
+ * @param {string} workflowRunLink - Link to the workflow run
+ * @param {string} docsWorkflowRunUrl - Link to the docs workflow run
+ * @param {string} mainMessage - Main message content
+ * @returns {string} Comment template
+ */
+function getCommentTemplate(commitSha, workflowRunLink, docsWorkflowRunUrl, mainMessage) {
+  return `
+Documentation preview for ${commitSha} ${mainMessage}
+
+<details>
+<summary>More info</summary>
+
+- Ignore this comment if this PR does not change the documentation.
+- It takes a few minutes for the preview to be available.
```

Comment:
Also this line no longer reflects the current implementation, we create the comment after the deploy. (We removed the pre-build one.)
```suggestion
```